src/modules/appointments/appointments.service.ts

```diff
@@ -0,0 +1,139 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { CreateAppointmentInput } from './dto/create-appointment.input';
+import { Appointment } from './entities/appointment.entity';
+
+@Injectable()
+export class AppointmentsService {
+  constructor(
+    @InjectRepository(Appointment)
+    private appointmentsRepository: Repository<Appointment>,
+  ) {}
+  async create(
+    createAppointmentInput: CreateAppointmentInput,
+  ): Promise<Appointment> {
+    try {
+      //* Getting all appointments in this room
+      const appointmentsInThisRoom = await this.findManyWithOptions({
+        where: { roomId: createAppointmentInput.roomId },
+      });
+
+      //* Getting time of this appointments and sort them
```

Comment:
You have a lot of logic below, looks like it can be simplified. @denismukhopadov could advise how to do this.